src/Aspire.Hosting/Dcp/DcpPipelineBuilder.cs

```diff
@@ -10,24 +10,24 @@ namespace Aspire.Hosting.Dcp;
 
 internal static class DcpPipelineBuilder
 {
-    public static ResiliencePipeline BuildDeleteRetryPipeline(ILogger logger)
+    public static ResiliencePipeline<bool> BuildDeleteRetryPipeline(ILogger logger)
     {
-        var ensureDeleteRetryStrategy = new RetryStrategyOptions()
+        var ensureDeleteRetryStrategy = new RetryStrategyOptions<bool>()
         {
             BackoffType = DelayBackoffType.Exponential,
             Delay = TimeSpan.FromMilliseconds(200),
             UseJitter = true,
             MaxRetryAttempts = 10, // Cumulative time for all attempts amounts to about 15 seconds
             MaxDelay = TimeSpan.FromSeconds(3),
-            ShouldHandle = new PredicateBuilder().Handle<Exception>(),
+            ShouldHandle = args => ValueTask.FromResult(!args.Outcome.Result),
```

Comment:
What about when an exception occurs that isn't Http NotFound? Should we retry on that?

Reply:
No, the retry here is specifically just for not found.